trajectory_processing/src/iterative_time_parameterization.cpp

```diff
@@ -111,13 +112,18 @@ void IterativeParabolicTimeParameterization::applyVelocityConstraints(robot_traj
   {
     const robot_state::RobotStatePtr &curr_waypoint = rob_trajectory.getWayPointPtr(i);
     const robot_state::RobotStatePtr &next_waypoint = rob_trajectory.getWayPointPtr(i+1);
-    
+
+    double velocity_scaling_factor = 1.0;
+
+    if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+      velocity_scaling_factor = max_velocity_scaling_factor;
```

Comment:
I concur with @skohlbr -- no assert, as 0.0 should be properly handled (we really don't want another instance of the "is_diff" nightmare where we are constantly having issues with default values of ROS messages). I would suggest that we only warn for values < 0 and greater than 1 if we want a warning.